src/components/FormFields/DataIntegrityChecksField.js

```diff
@@ -0,0 +1,158 @@
+import React, { useCallback, useState } from 'react'
+import { PropTypes } from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import {
+    FieldGroup,
+    Radio,
+    Transfer,
+    TransferOption,
+    ReactFinalForm,
+    InputFieldFF,
+    Help,
+} from '@dhis2/ui'
+import cx from 'classnames'
+import { hooks } from '../Store'
+import {
+    getCheckName,
+    severityMap,
+} from '../../services/server-translations/dataIntegrityChecks'
+import styles from './DataIntegrityChecksField.module.css'
+
+const { Field, useField } = ReactFinalForm
+
+const VALIDATOR = value =>
+    value && value.length < 1
+        ? i18n.t('Please select checks to run.')
+        : undefined
+
+const DataIntegrityChecksField = ({ label, name }) => {
+    const options = hooks.useParameterOptions('dataIntegrityChecks')
+    const {
+        input: { value, onChange },
+    } = useField(name)
+
+    const hasValue = !!value && value.length > 0
+    const [runSelected, setRunSelected] = useState(hasValue)
+
+    const translatedOptions = options
+        .map(option => ({
+            ...option,
+            value: option.name,
+            label: getCheckName(option.name),
+            severity: severityMap[option.severity],
+        }))
+        .sort((a, b) => a.label.localeCompare(b.label))
+
+    const toggle = ({ value }) => {
+        const checked = value === 'true'
+
+        if (!checked) {
+            // clear checks when "Run all" is selected
```

Comment:
We have destroyOnUnregister enabled, see [here](https://github.com/dhis2/scheduler-app/blob/6915b1858a13bcfaf4486edd76cd1789a3871ff6/src/components/Forms/JobEditFormContainer.js#L50), which should take care of removing the value if the field is unregistered. But I see you linked an open issue about conditional rendering causing issues. We're rendering the parameter fields conditionally, could what you encountered be solved in a similar manner?

Reply:
Well, I got errors that crashed the app, so I added the `hidden`-prop to fix this. It wasn't just because of conditional rendering, but conditional rendering in combination with updating the state of the component.

Reply:
The error I encountered was ```Cannot update a component (`DataIntegrityChecksField`) while rendering a different component (`ForwardRef(Field)`). To locate the bad setState() call inside `ForwardRef(Field)`,```

Reply:
I see. It seems like you're initializing the state for `runSelected` based on the previous value of the field. Which will be there if the user toggles back and forth between "run all" and "run only". It'll be lost as soon as the user switches to another job type. If you lose that and initialize `runSelected` to `false` there's no longer a need to clear the onChange (provided you conditionally render the transfer field in the field group). That means you shouldn't run into the linked bug either (at least I didn't during testing, but let me know if I missed it).

I'd go with that. We don't save state for parameter settings between jobs anyway, and if not saving it for this section doesn't require workarounds and simplifies things I would personally go in that direction.

Reply:
Haha, oh yeah of course you're right. Minor oversight on my part 😅. Wait, let me check one thing that might still satisfy that requirement.

Reply:
Ok. So it seems that using useField prevents the field from unregistering. That's why you need to clear manually even if you're no longer rendering the field.

Using useFormState allows access to the form values without keeping the field registered:

```
    const { values } = useFormState({ subscription: { values: true } })
    const value = values?.jobParameters?.checks

    // ...
    
    const toggle = ({ value }) => {
        const checked = value === 'true'
        setRunSelected(checked)
    }

    // ...

            {runSelected && <Field
                name={name}
                component={ChecksTransfer}
                options={translatedOptions}
                label={label}
                validate={VALIDATOR}
            />}
```

Accessing the field from FF state might get a little ugly because the name uses final form's notation for nested fields. But something like this would keep the desired behaviour and allow you to do conditional rendering without encountering the bug (at least I didn't with this approach).

Reply:
Thanks for looking into this! Actually makes sense for `useField` to prevent unregistering of the field. However, I don't really think this is much better than the current solution? This is a bug in `react-final-form` anyway, and it's referenced in the code - so once it's fixed we could move the "hidden"-prop and conditionally render instead. I don't really see the benefit here - you would even have more re-renders due to subscribing to all values (I know that the current code also re-renders for each change, but that is easily preventable with `react.memo()` if we want that). 

> I do wonder if our usage of initialValues in the app is a little flawed. They'll be lost as soon as final form unregisters a field and clears the value, like when a user switches jobTypes for an existing job.

Yeah, haven't looked too much into that, but seems like we would need to change the `destroyOnUnmount` or something if we would like to keep the initialValues when unmounting fields?

Reply:
> Yeah, haven't looked too much into that, but seems like we would need to change the destroyOnUnmount or something if we would like to keep the initialValues when unmounting fields?

Yeah, yeah the reason that that's there is so that if you switch a jobType, the existing parameters don't persist. Otherwise those parameters will be sent to the backend if the user submits. I suppose the backend would probably drop unrecognized job parameters, but it seemed more correct to me at the time.

That's also why I was thinking of a way to automatically remove the transfer values if it unregisters. My assumption was that selecting run all would mean you don't need the checks job parameter at all. But of course, run all is identical to selecting all checks (I assume?). Which is obvious of course, but I'd not even stopped to consider that.

So, first off, I'll approve the PR since I don't want to block you. Since I rewrote a lot of this app I tend to get a little product-managery about it, but that's not actually my position.

That being said. I can't help myself thinking that the design could be simplified. Instead of a "run all" radio button I'd just render the transfer at all times and forego the radios. After all, the transfer already has a convenient "select all" control. That would simplify our code _and_ the UI if you ask me. Do you know if there's a technical reason why we didn't go with that approach?

Reply:
> So, first off, I'll approve the PR since I don't want to block you. Since I rewrote a lot of this app I tend to get a little product-managery about it, but that's not actually my position.

I absolutely understand this, no worries! There's a lot of decision that's taken here, and I don't know the full background. I've just tried to implement this feature without too drastic changes. 

> That being said. I can't help myself thinking that the design could be simplified. Instead of a "run all" radio button I'd just render the transfer at all times and forego the radios. After all, the transfer already has a convenient "select all" control. That would simplify our code and the UI if you ask me. Do you know if there's a technical reason why we didn't go with that approach?

I do agree with this point, and this was my first implementation, with a "If no checks are selected, all checks will be run"-message in the "SelectedEmptyComponent". But I talked to Joe, and he proposed the implemented solution. The reason being that it might be a bit confusing that actually selecting nothing selects all, and the reason below:
Manually having to select all would also probably not be the best, since if new checks are added in the backend, you would have to manually select them again. In the current solution, all checks will run regardless of user interaction after the fact.

Reply:
I see. Since we already talked a bit on slack, I'll just add here what I said there: seems tricky to really simplify the UI then on our end. The requirements kind of dictate the current solution. Would be nice if eventually we could get rid of the two types of "all checks", because now effectively we have "all current and future checks" and "all explicitly selected checks". To me that wasn't really clear from the UI. Anyway, something for the future I guess. Thanks for explaining!